src/components/chat-item/chat-wrapper.ts

```diff
@@ -28,15 +30,27 @@ export class ChatWrapper {
   private readonly intermediateBlockContainer: ExtendedHTMLElement;
   private readonly promptInputElement: ExtendedHTMLElement;
   private readonly promptInput: ChatPromptInput;
+  private readonly footerSpacer: ExtendedHTMLElement;
+  private readonly headerSpacer: ExtendedHTMLElement;
   private readonly promptInfo: ExtendedHTMLElement;
   private readonly promptStickyCard: ExtendedHTMLElement;
+  private tabHeaderDetails: ExtendedHTMLElement;
+  private tabModeSwitchTimeout: ReturnType<typeof setTimeout> | null;
   private lastStreamingChatItemCard: ChatItemCard | null;
   private lastStreamingChatItemMessageId: string | null;
   private allRenderedChatItems: Record<string, ChatItemCard> = {};
   render: ExtendedHTMLElement;
   constructor (props: ChatWrapperProps) {
     this.props = props;
-    MynahUITabsStore.getInstance().getTabDataStore(this.props.tabId).subscribe('chatItems', (chatItems: ChatItem[]) => {
```

Comment:
I'm confused, what is the difference between `subscribe()` and `addListenerToDataStore`? Or is this a renamed function?

Reply:
They're doing the exact thing. However keep it consistent through the whole usage, preferred way is `addListenerToDataStore`. Actually `subscribe` should be private.

Reply:
So why do we have both if they do the exact same thing?

Reply:
One of them is one level deep, the other one is directly on the tabStore object.

So: `tabStore. addListenerToDataStore` and `tabStore.getTabStoreData().subscribe`. 
For example If you already grabbed the tabStoreData before, you can use it directly.
```typescript
const storeData = MynahUITabsStore.getInstance().getTabDataStore(tabId);
// do some stuff with storeData
storeData.subscribe(...)
```

But if you directly need to listen a value change on the store:
```typescript
 MynahUITabsStore.getInstance().addListenerToDataStore(...)
```